wordle.py

```diff
@@ -0,0 +1,97 @@
+import random
+from constants import GREEN, YELLOW, DEFAULT, CURSOR_UP, LINE_CLR
+
+
+def get_random_word(filename):
+    '''
+    Gets a random word from filename, which is a sequence of words
+    separated by newlines
+    '''
+    with open(filename, 'r') as f:
+        words = f.readlines()
+
+    return random.choice(words).strip()
+
+
+def cursor_up_line_clear(n=1):
+    '''
+    Moves the terminal cursor n lines up and
+    removes any existing data on the lines
+    '''
+    print((CURSOR_UP + LINE_CLR)*n, end="")
+
+
+def validate(guess, status):
+    '''
+    Checks if a guess is valid i.e has proper length and
+    consists of alphabets only
+
+    Also updates status properly in case of error
+    '''
+
+    if len(guess) != 5:
+        status['err_msg'] = 'Please make a guess that is 5 characters long'
+        return False
+    elif not guess.isalpha():
+        status['err_msg'] = 'Non Alphabetic guesses not allowed'
+        return False
+    else:
+        return True
+
+
+def take_guess():
+    '''
+    Prompts the user for a guess until they
+    enter a valid guess (5 English alphabets)
+    '''
+
+    bad_input = False
+    guess = input()
+
+    # Represents the state of the guess
+    status = {
+        'err_msg': None
+    }
```

Comment:
Why we need a dict here?

Reply:
Code refactored to not need a dict here.